src/lilURL.php

```diff
@@ -61,13 +69,46 @@ public function handleRedirect($id)
         if ($id != '' && $id != basename($_SERVER['PHP_SELF']) && $id != '?login') {
             $location = $this->getURL($id);
             if ($location != false) {
+                $this->trackHit($id);
```

Comment:
It's a bit complicated to explain as I think people confuse async with some action that is disconnected from the php request. Suffice it to say that the page will _always_ wait for the request to finish before the response is fully sent to the browser. The async part just means delay the request until the shutdown of the php request or until it is explicitly called.